### PR TITLE
 Add more information about event data to 'hook add' API function documentation.

### PR DESCRIPTION
**Pull request by ywkaras**
This PR is dependent on https://github.com/apache/trafficserver/pull/4576 .

### CODE REVIEW

**Comment by ywkaras** [comment]
[approve ci autest]

File: doc/developer-guide/api/functions/TSHttpHookAdd.en.rst [hunk 1]
```diff
@@ -77,9 +77,11 @@ function will be TS_EVENT_HTTP_READ_REQUEST_HDR.
 When a continuation is triggered by a hook, the actual type of the event data
 (the void pointer passed as the third parameter to the continuation function) is
 determined by which hook it is.  For example, for the hook ID TS_HTTP_TXN_CLOSE_HOOK,
-the event data is of type TSHttpTxn.  This is the case regardless of whether the
+the event data is of type :type:`TSHttpTxn`.  This is the case regardless of whether the
 continuation was added to the hook using :func:`TSHttpTxnHookAdd`, :func:`TSHttpSsnHookAdd`
-or :func:`TSHttpHookAdd`.
+or :func:`TSHttpHookAdd`.  If the event data is of type :type:`TSHttpTxn`, :type:`TSHttpSsn` or
+:type:`TSVConn`, the continuation function can assume the mutex of the indicated
+event data object is locked.  (But the continuation function should not unlock it.)
```
Comment (SolidWallOfCode) [inline doc/developer-guide/api/functions/TSHttpHookAdd.en.rst]: Change to "must **not** unlock the mutex".

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by ywkaras** [comment]
[approve ci autest]

File: tests/tools/plugins/test_hooks.cc [hunk 2]
```diff
@@ -288,6 +325,22 @@ TSPluginInit(int argc, const char *argv[])
...
+  // NOTE: as of January 2019 these two hooks are only triggered for TLS connections.  It seems that, at trafficserver
+  // startup, spurious data on the TLS TCP port may cause trafficserver to attempt (and fail) to create a TLS
+  // connection.  If this happens, it will result in TS_VCONN_START_HOOK being triggered, and then TS_VCONN_CLOSE_HOOK
+  // will be triggered when the connection closes due to failure.
+  //
+  TSHttpHookAdd(TS_VCONN_START_HOOK, gCont);
+  TSHttpHookAdd(TS_VCONN_CLOSE_HOOK, gCont);
+
+  // TSHttpHookAdd(TS_SSL_SESSION_HOOK, gCont); -- Event is TS_EVENT_SSL_SESSION_NEW -- Event data is TSHttpSsn
+  // TSHttpHookAdd(TS_SSL_SERVER_VERIFY_HOOK, gCont);
```
Comment (bryancall) [inline tests/tools/plugins/test_hooks.cc]: Is there a reason these are commented out?  Why are the hooks not being called?

**Comment by bryancall** [comment]
@ywkaras Can you please squash the commits.

File: tests/tools/plugins/test_hooks.cc (hunk 2 above)
Comment (ywkaras) [reply tests/tools/plugins/test_hooks.cc]: They don't seem to be working.

**Comment by SolidWallOfCode** [comment]
@ywkaras - did you mean to drop all those test changes in your recent force push?

**Comment by ywkaras** [comment]
@SolidWallOfCode they we're dropped, they were part of another PR that has been merged.

**Comment by ywkaras** [comment]
[approve ci autest]

**Pull request merged**